src/librustdoc/html/markdown.rs

```diff
@@ -531,6 +530,7 @@ extern {
     fn hoedown_buffer_new(unit: libc::size_t) -> *mut hoedown_buffer;
     fn hoedown_buffer_puts(b: *mut hoedown_buffer, c: *const libc::c_char);
     fn hoedown_buffer_free(b: *mut hoedown_buffer);
+    fn hoedown_buffer_put(b: *mut hoedown_buffer, c: *const libc::c_char, len: libc::size_t);
```

Comment:
The second parameter is [`const uint8_t *data`](https://github.com/rust-lang/hoedown/blob/da282f1bb7277b4d30fa1599ee29ad8eb4dd2a92/src/buffer.h#L75) so here it should be `*const u8`. That also means ` as *const libc::c_char` can be removed from below.